consensus/block.go

```diff
@@ -0,0 +1,117 @@
+package consensus
+
+import (
+	"github.com/google/uuid"
+	"github.com/spacemeshos/go-spacemesh/mesh"
+	"time"
+)
+
+type BlockID uint32
+type LayerID uint32
+
+var layerCounter LayerID = 0
+
+
+
+type TortoiseBlock struct {
+	Id         BlockID
+	LayerIndex LayerID
+	Data       []byte
+	Coin       bool
+	Timestamp  time.Time
+	ProVotes   uint64
+	ConVotes   uint64
+	BlockVotes map[BlockID]bool
+	ViewEdges  map[BlockID]struct{}
+}
+
+func (b TortoiseBlock) ID() BlockID {
+	return b.Id
+}
+
+func (b TortoiseBlock) Layer() LayerID {
+	return b.LayerIndex
+}
+
+
+
+func NewBlock(coin bool, data []byte, ts time.Time, layerId LayerID) *TortoiseBlock {
+	b := TortoiseBlock{
+		Id:         BlockID(uuid.New().ID()),
+		LayerIndex: layerId,
+		BlockVotes: make(map[BlockID]bool),
```

Comment:
vote patterns for different layers are supposed to be encoded here
so cant be map[BlockID]bool

Reply:
this is the old impl of the tortoise, new implementation will override this probably